drafts/2019-05-16-restful-config-viewing-and-validation-service.adoc

```diff
@@ -0,0 +1,327 @@
+---
+layout: post
+title: "Testing database connections in Open Liberty apps with REST APIs"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/u/30153613
+author_github: https://github.com/nmittles
+seo-title: Testing database connections in Open Liberty apps with REST APIs - OpenLiberty.io
+seo-description: Looking for an easy and fast way to test your database connections? You can now validate your connections by invoking REST endpoints provided by the configValidator 1.0 beta feature. In addition to being easy and fast, these REST endpoints will exercise the same code paths as your applications, giving you confidence in your server configuration.
+blog_description: "Looking for an easy and fast way to test your database connections? You can now validate your connections by invoking REST endpoints provided by the configValidator 1.0 beta feature. In addition to being easy and fast, these REST endpoints will exercise the same code paths as your applications, giving you confidence in your server configuration."
+---
+= Testing database connections in Open Liberty apps with REST APIs
+Nathan Mittlestat <https://github.com/nmittles>
+
+== Testing database connections in Open Liberty apps with REST APIs
+
+Looking for an easy and fast way to test your database connections? You can now validate your connections by invoking REST endpoints provided by the configValidator 1.0 beta feature. In addition to being easy and fast, these REST endpoints will exercise the same code paths as your applications, giving you confidence in your server configuration. 
+
+By itself, configValidator 1.0 allows viewing of a server's current configuration via the `/ibm/api/config` endpoint. REST endpoints capable of validating specific resources become available under the `/ibm/api/validation` parent endpoint, when the `configValidation-1.0` feature is combined with features implementing configuration validation. Validation of JDBC, JCA, Cloudant, and JMS configuration elements are currently supported. Looking for more validation support? Let us know in the comments!
+
+Both the validation and config endpoints are generated using a combination of the config element name, and unique ID:
+```code
+/ibm/api/config/{configElementName}/{uid}
+/ibm/api/validation/{configElementName}/{uid}
+```
+Now, let’s take a look at how you can use this new REST API to validate your configuration.
+
+Here is a sample server.xml configuration that utilizes configValidator-1.0, jdbc-4.2, and jca-1.7.
+
+```xml
+<server>
+  <featureManager>
+    <feature>appSecurity-2.0</feature>
```

Comment:
is security needed always? I'm not sure. If it isn't, we should eliminate it from this first example

Reply:
Removed.  It looks to have been a holdover from a test I copied and pasted from.